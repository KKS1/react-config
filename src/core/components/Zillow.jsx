import React, { Component } from "react";
import styled from "styled-components";
import ZillowService from "../services/zillowService";

const formatter = new Intl.NumberFormat('en-US', {
  style:    'currency',
  currency: 'USD',
});

const ZillowLogoLink = styled.a`
  height: 3.125rem;
  width: 3.125rem;
`;

const ZillowImage = styled.img`
  height: 3.125rem;
  width: 12.5rem;
`;

class Zillow extends Component {
  constructor(props) {
    super(props)
    this.zillowService = new ZillowService()
  }

  fillZillowData = async () => {
    const {onChange, formData} = this.props
    const zillowData = await this.zillowService.search();

    if( !zillowData ) return;

    const { zestimate = '' , homeDetailsUrl, zillowId, address }  = zillowData;
    const formattedZestimate = formatter.format(zestimate).replace('.00', '');

    onChange(
      {
        ...formData, ...{zestimate: formattedZestimate, homeDetailsUrl, zillowId, address}
      }
    )
  }

  componentDidMount() {
    this.fillZillowData()
  }

  render() {
    const {formData, formData: {zillowId, zestimate, homeDetailsUrl, address} = {}} = this.props;

    return (
      <div>
        { (zillowId && zestimate) ?
          <div>
            <div className='row'>
              <div className='col-sm-6'>
                <div>
                  Zestimate - {zestimate}
                </div>
                <div className="zillow-external-link">
                  <a href={homeDetailsUrl} target="_blank">
                    See more details for {address} on Zillow
                    <i className="fa c-external-link" />
                  </a>
                </div>
              </div>
              <div className='col-sm-6'>
                <ZillowLogoLink href="https://www.zillow.com/">
                  <ZillowImage src="http://www.zillow.com/widgets/GetVersionedResource.htm?path=/static/logos/Zillowlogo_200x50.gif" alt="Zillow Real Estate Search"/>
                </ZillowLogoLink>
              </div>
            </div>
          </div>
          : null
        }
      </div>
    )
  }
}

export default Zillow
