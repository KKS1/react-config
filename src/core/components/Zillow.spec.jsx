import React, { Component } from "react";
import Zillow from "./Zillow";

describe("Zillow", () => {
  test("render_Zillow", () => {
    const wrapper = shallow(<Zillow />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("attach_props", () => {
    const props = {
      zestimate: '103948954',
      zillowId: '103293',
      homeDetailsUrl: 'https://www.zillow.com',
      address: '1568 Alabama St.'
    };
    const wrapper = shallow(<Zillow {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });
});
