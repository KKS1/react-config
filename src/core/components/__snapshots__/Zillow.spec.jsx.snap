// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zillow attach_props 1`] = `
Zillow {
  "context": Object {},
  "fillZillowData": [Function],
  "props": Object {
    "address": "1568 Alabama St.",
    "homeDetailsUrl": "https://www.zillow.com",
    "zestimate": "103948954",
    "zillowId": "103293",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <Zillow
        address="1568 Alabama St."
        homeDetailsUrl="https://www.zillow.com"
        zestimate="103948954"
        zillowId="103293"
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div />,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  "zillowService": ZillowService {
    "search": [Function],
  },
}
`;

exports[`Zillow render_Zillow 1`] = `
Zillow {
  "context": Object {},
  "fillZillowData": [Function],
  "props": Object {},
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <Zillow />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div />,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  "zillowService": ZillowService {
    "search": [Function],
  },
}
`;
